Code.gs

```diff
@@ -1,69 +1,206 @@
+    if (h == DocumentApp.ParagraphHeading.HEADING1) {
+      // Set the next paragraph to heading 2 and add the Trello Card Title URL link
+      var np = pars[Number(i) + 1]
+      var nptext = np.getText()
+      Logger.log(nptext)
+      np.setHeading(DocumentApp.ParagraphHeading.HEADING2)
+
+      //Store the text as the Trello Card Title
+      var trelloCardTitle = nptext
       
-      //check if the company name matches the orgName
-      var orgNameSearch = row[CompanyCol]
-      if (orgNameSearch === orgName) {
-        timesheetUrl = row[TimesheetCol]
-        Logger.log(timesheetUrl)
-        return timesheetUrl
+      //Get the Url from the card title, add the link to the journal
+      var trelloCardUrl = getTrelloCardUrl(trelloBoardId, trelloCardTitle)
+      
+      if (trelloCardUrl === null) {
+      Logger.log('Trello card Not found in trello board')
       }
-    })
-    
-    Logger.log(timesheetUrl)
+      
+      // Add the Hyperlink to the journal
+      np.setLinkUrl(trelloCardUrl)
+      
+     }
+  }
   
-    //Open timesheet and add the Trello Card and Hyperlink 
-    //var timesheetTask = SpreadsheetApp.openById(timesheetId).getSheetByName('Timesheet').getRange(timesheetrange)
-    if (timesheetUrl === '') {
-      Logger.log('No Timesheet URL found in Org Spreadsheet')
-    } else {
+  //Open the Timesheet spreadsheet
+  var tsData = SpreadsheetApp.openByUrl(timesheetUrl)
+    .getSheetByName('Timesheet')
+    .getRange("A1:A").getValues()
+  //Get the last row with data in the Date column, use this to add the Trello Card and 
+  //Journal URL link to the timesheet
+  var lastTsRow = tsData.filter(String).length 
+  Logger.log(lastTsRow)
+  
+  //Open timesheet and add the Hyperlink to Journal
+  if (timesheetUrl === null) {
+
+    Logger.log('No Timesheet URL found in Org Spreadsheet')
+    return
+    
+  } else {
               
-      var timesheetTask = SpreadsheetApp.openByUrl(timesheetUrl).getSheetByName('Timesheet').getRange(timesheetrange)
-      timesheetTask.setFormula('=HYPERLINK("' + trellourl + '", " ' + trelloCardName + '")')
-    }
+    var timesheetTask = SpreadsheetApp.openByUrl(timesheetUrl)
+      .getSheetByName('Timesheet')
+      .getRange(lastTsRow, TIMESHEET_COL_)
+    timesheetTask.setFormula('=HYPERLINK("' + DocUrl + '", " ' + trelloCardTitle + '")')
+    
+  }
+    
+return SUCCESS
+
+}
+
+function getTrelloCardUrl(trelloBoardId, trelloCardTitle) {
+  
+  //Get the JSON response of the Trello Board
+  var API_KEY = PropertiesService.getScriptProperties().getProperty("API_KEY")
+  var TOKEN = PropertiesService.getScriptProperties().getProperty("TOKEN")
+  var url = "https://api.trello.com/1/boards/" + trelloBoardId + "/cards/?fields=name,url&key=" + API_KEY + "&token=" + TOKEN
+  var response = UrlFetchApp.fetch(url).getContentText()
+  var cardUrl = null
+  
+  var obj = JSON.parse(response)
+        
+  //Find the card title in the list of board cards   
+  for (x in obj) {
```

Comment:
Could we avoid single letter variables apart from in test code. "key" is typically used here.